src/taskProvider.ts

```diff
@@ -0,0 +1,68 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+'use strict';
+
+import * as serverUtils from './omnisharp/utils';
+import * as vscode from './vscodeAdapter';
+
+import { OmniSharpServer } from './omnisharp/server';
+import { TaskRevealKind } from 'vscode';
+
+export function activate(_context: vscode.ExtensionContext, server: OmniSharpServer): vscode.Disposable {
+	if (!vscode.workspace.workspaceFolders) {
+		return new vscode.Disposable(() => {});
+	}
+
+	return vscode.workspace.registerTaskProvider('dotnet', {
+		provideTasks: () => {
+			return provideDotnetTasks(server);
+		},
+		resolveTask(_task: vscode.Task): vscode.Task | undefined {
+			return undefined;
+		}
+	});
+}
+
+async function provideDotnetTasks(server: OmniSharpServer): Promise<vscode.Task[]> {
+	let tasks: vscode.Task[] = [];
+	
+	let workspaceInformation = await serverUtils.requestWorkspaceInformation(server);
+	
+	if (workspaceInformation.MsBuild && workspaceInformation.MsBuild.Projects) {
+		for (let project of workspaceInformation.MsBuild.Projects) {
+			tasks.push(provideBuildTask(project.AssemblyName, project.Path));
+		}
+	}
+
+	if (workspaceInformation.DotNet && workspaceInformation.DotNet.Projects) {
+		for (let project of workspaceInformation.DotNet.Projects) {
+			tasks.push(provideBuildTask(project.Path, project.Path));
+		}
+	}
```

Comment:
FWIW, I wouldn't bother adding this feature for project.json, since the support is officially deprecated.